build.gradle

```diff
@@ -76,4 +76,78 @@ project(':configurator') {
                     'Main-Class': mainClassName)
         }
     }
-}
\ No newline at end of file
+}
+ext {
+    systemNames = ["Linux","Windows","Mac"]
+    jreDir = new File("$buildDir/jre.tar.gz")
+    jarDir = new File("$buildDir/libs")
+}
+
+import java.security.MessageDigest
+task downloadJre  {
+    def originalFileHash = "dd28d6d2cde2b931caf94ac2422a2ad082ea62f0beee3bf7057317c53093de93"
+    // Download JRE
+    doFirst {
+        if(!jreDir.exists()) {
+            def downloadUrl = new URL("https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk8u212-b03/OpenJDK8U-jdk_x64_linux_hotspot_8u212b03.tar.gz")
+            def file = new FileOutputStream(jreDir)
+            def out = new BufferedOutputStream(file)
+            try{
+                out << downloadUrl.openStream()
+                out.close()
+            }
+            catch(IOException e){
+                e.printStackTrace()
+            }
+            finally {
+                file.close()
+            }
+        }
+    }
+    // Check the hash of JRE
+    doLast {
+        def digest = MessageDigest.getInstance("SHA-256");
+        def buffer = new byte[1024]
+        def len
+        def inputStream = new FileInputStream(jreDir)
+        try {
+            while ((len = inputStream.read(buffer, 0, 1024)) != -1) {
+                digest.update(buffer, 0, len)
+            }
+            BigInteger bigInt = new BigInteger(1, digest.digest())
+            String hashValue = bigInt.toString(16)
+            if(!hashValue.equals(originalFileHash)){
+                throw new Exception("Please check if JRE is downloaded securely")
+            }
+        } catch (IOException e) {
+            e.printStackTrace()
+        } finally {
+            inputStream.close()
+        }
+    }
```

Comment:
Similar to the `doFirst` let's use the language:

```groovy
        final filename = 'OpenJDK8U-jdk_x64_linux_hotspot_8u212b03.tar.gz'
        def digest = MessageDigest.getInstance('SHA-256')
        new File( "${jreDir.getAbsolutePath()}/${filename}" as String ).eachByte( 1024 ) { buf, len ->
            digest.update( buf, 0, len )
        }
        final hash = digest.digest().encodeHex() as String

        if( hash != originalFileHash ) {
            throw new RuntimeException("Hash mismatch for downloaded JRE. File may have been modified.")
        }
```

Again though note I've just shoe horned in the `filename` - you'll need to externalise.